src/coreclr/tools/aot/ObjWriter/build.sh

```diff
@@ -37,8 +39,13 @@ PatchApplied=0
 
 if [ ! -d llvm-project ]; then
     # Clone the LLVM repo
-    git clone --depth 1 -b release/12.x https://github.com/llvm/llvm-project.git || exit 1
-    cd llvm-project/llvm || exit 1
+    git clone --no-checkout --depth 1 -b $LLVMBranch https://github.com/llvm/llvm-project.git || exit 1
+    cd llvm-project || exit 1
+    # Purposefuly ignoring exit codes of sparse-checkout so that this works with git lower than 2.26
+    git sparse-checkout init
+    git sparse-checkout set /llvm/\* !/llvm/test/\*
+    git checkout $LLVMBranch || exit 1
```

Comment:
Not worth restarting the build, but it is a good practice to always quote variables:
```suggestion
    git clone --no-checkout --depth 1 -b "$LLVMBranch" https://github.com/llvm/llvm-project.git || exit 1
    cd llvm-project || exit 1
    # Purposefuly ignoring exit codes of sparse-checkout so that this works with git lower than 2.26
    git sparse-checkout init
    git sparse-checkout set /llvm/\* !/llvm/test/\*
    git checkout "$LLVMBranch" || exit 1
```